### PR TITLE
Add double click action

### PR DESCRIPTION
**Pull request by joshgordon**
This PR adds the ability to trigger double-clicks on elements.

### CODE REVIEW

File: test/unit/lib/action.test.js [hunk 1]
```diff
@@ -233,7 +242,50 @@ describe('lib/action', () => {
 
 			it('clicks the specified element on the page', () => {
 				expect(puppeteer.mockPage.click).toHaveBeenCalledTimes(1);
-				expect(puppeteer.mockPage.click).toHaveBeenCalledWith(clickElementMatches[2]);
+				expect(puppeteer.mockPage.click).toHaveBeenCalledWith(clickElementMatches[3], {});
```
Comment (danyalaytekin) [inline test/unit/lib/action.test.js:245]: Since I've changed the payload above (sorry):

```suggestion
				expect(puppeteer.mockPage.click).toHaveBeenCalledWith(clickElementMatches[3], {
					clickCount: 1
				});
```

File: test/unit/lib/action.test.js [hunk 2]
```diff
@@ -233,7 +242,50 @@ describe('lib/action', () => {
...
+			let clickElementMatches;
+			let clickElementResolvedValue;
+
+			beforeEach(async () => {
+				global.document.querySelector.mockReturnValue(mockElement);
+				clickElementMatches = 'double click element foo'.match(clickElementAction.match);
+				clickElementResolvedValue = await clickElementAction.run(puppeteer.mockBrowser, puppeteer.mockPage, {}, clickElementMatches);
+			});
+
+			it('clicks the specified element on the page', () => {
```
Comment (danyalaytekin) [inline test/unit/lib/action.test.js:286]:
```suggestion
			it('double clicks the specified element on the page', () => {
```

**Review by danyalaytekin — COMMENTED** [review]
Hi @joshgordon, thank you for proposing this new action and for updating the unit tests too.

Was there a particular scenario that you were trying to test, and couldn't... text selection maybe?  I'd be interested in hearing about it.  Or if this is more just an altruistic effort to expand the portfolio of Pa11y Actions that's great too.

I've left a few low level comments inline - please let me know your thoughts.

Finally, I'm thinking it could be good to add an integration test, the motivation to do this increasing when I spotted [this Stack Overflow comment](https://stackoverflow.com/questions/64384430/how-to-double-click-in-puppeteer#comment118361106_64384883), even if its content no longer applies or came from a misunderstanding.

Here are some example files for our integration test for `click`:
- [expectations](https://github.com/pa11y/pa11y/blob/main/test/integration/cli/actions-click-element.test.js)
- [html](https://github.com/pa11y/pa11y/blob/main/test/integration/mock/html/actions-click-element.html)

Thanks again Josh.  Any questions please ask.

**Comment by joshgordon** [comment]
> Was there a particular scenario that you were trying to test, and couldn't... text selection maybe?

We have an ext.js application that expects double-clicks on data-rows to view/change the remainder of the properties of the object described by the row. I played around with clicking multiple times but none of them got the modal to pop up, and finally came to the conclusion that I just needed to patch it in myself. 

I went ahead and added all the suggestions made. I'll try to find a couple cycles to put together an integration test - that looks straightforward enough.

**Comment by danyalaytekin** [comment]
Currently bringing tests back into sync. May be some chop.
